Subagent extension fails to load (bundled-extension-paths.js) + orphaned bg processes between auto-mode units

## Problem

Two related lifecycle bugs in auto-mode: (1) the subagent extension fails to load because its relative import `../../../bundled-extension-paths.js` resolves against `~/.gsd/` (where the extension is resource-copied) instead of the npm package `dist/` directory, and (2) background processes (e.g. Vite dev servers) spawned during UAT/validation units are not cleaned up between unit sessions, causing port conflicts, stuck loops, and timeouts on subsequent units.

## Environment
- GSD version: 2.28.0
- Model: claude-sonnet-4-6 / claude-opus-4-6[1m]
- Unit: multiple — 12 units dispatched twice (stuck-loop), errors in complete/milestone-M031, complete/slice-M031-S02, execute/task-M031-S02-T01

## Reproduction Context

**Bug 1 (extension load):** Occurs on every `gsd` startup. The subagent extension at `~/.gsd/agent/extensions/subagent/index.ts` imports `parseBundledExtensionPaths` from `../../../bundled-extension-paths.js`. When jiti compiles this from the resource-copied location, the relative path resolves to `~/.gsd/bundled-extension-paths.js` which does not exist. The actual module is at `gsd-pi/dist/bundled-extension-paths.js`. Error:
```
[gsd] Extension load error: Failed to load extension: Cannot find module '../../../bundled-extension-paths.js'
Require stack: ~/.gsd/agent/extensions/subagent/index.ts
```

**Bug 2 (orphaned processes):** During auto-mode, execute-task and run-uat units spawn Vite dev servers via `bash` with `&` for browser-based verification. When the unit session ends and `dispatchNextUnit` creates a fresh session via `newSession()`, the OS-level background process is not terminated. The orphaned server keeps the port bound, causing subsequent units dev server launches to fail or conflict. `handleAgentEnd` and `dispatchNextUnit` contain no background process cleanup logic. This led to 12 stuck-loop anomalies and cost spikes (3.5x-4.8x average) from timeout recovery retries.

## Forensic Findings

- 12 `[WARNING] stuck-loop` anomalies across M025-M031 units (dispatched 2+ times each)
- 3 `[WARNING] cost-spike` anomalies ($6.47 research-slice, $3.32 run-uat, $4.11 validate-milestone)
- `edit` tool called without `path` parameter in complete/slice-M031-S02 and complete/milestone-M031 (model-level malformed tool call, self-recovered)
- Orphaned Vite process on port 5173 confirmed still running after auto-mode completed M031
- `verify-m031.py` referenced but not found on disk (`[Errno 2]`)
- `src/components/canvas/` referenced but not found

## Suggested Fix Area

**Bug 1:** `dist/resources/extensions/subagent/index.ts` — the relative import `../../../bundled-extension-paths.js` assumes the files position relative to `dist/` but extensions are resource-copied to `~/.gsd/agent/extensions/`. Either:
- Add `bundled-extension-paths` to jitis alias map in `packages/pi-coding-agent/dist/core/extensions/loader.js` (~line 238), or
- Copy `bundled-extension-paths.js` alongside the extension resources during `initResources()`, or
- Replace the relative import with a package-scoped import

**Bug 2:** `dist/resources/extensions/gsd/auto.ts` — `handleAgentEnd()` (~line 1150) and `dispatchNextUnit()` (~line 1990) have no mechanism to enumerate or kill background OS processes spawned by the prior units session. Options:
- Track PIDs of processes spawned via `bash` tool during each unit, kill on unit closeout
- Use `bg_shell` with `persist_across_sessions: false` (default) so the harness manages lifecycle
- Add a port-conflict check in `preDispatchHealthGate()` that kills processes on known dev ports before dispatch

---
*Auto-generated by `/gsd forensics`*

--- a/src/resources/extensions/bg-shell/process-manager.ts
+++ b/src/resources/extensions/bg-shell/process-manager.ts
@@ -375,6 +375,19 @@ export function cleanupAll(): void {
 	processes.clear();
 }
 
+/**
+ * Kill all alive, non-persistent bg processes.
+ * Called between auto-mode units to prevent orphaned servers from
+ * keeping ports bound across task boundaries (#1209).
+ */
+export function killSessionProcesses(): void {
+	for (const [id, bg] of processes) {
+		if (bg.alive && !bg.persistAcrossSessions) {
+			killProcess(id, "SIGTERM");
+		}
+	}
+}
+
 async function waitForProcessExit(bg: BgProcess, timeoutMs: number): Promise<boolean> {
 	if (!bg.alive) return true;
 	await new Promise<void>((resolve) => {

--- a/src/resources/extensions/gsd/auto-post-unit.ts
+++ b/src/resources/extensions/gsd/auto-post-unit.ts
@@ -202,10 +202,13 @@ export async function postUnitPreVerification(pctx: PostUnitContext): Promise<"d
       }
     }
 
-    // Prune dead bg-shell processes
+    // Prune dead bg-shell processes and kill non-persistent live ones.
+    // Without killing live processes between units, dev servers spawned during
+    // one task keep ports bound, causing conflicts in subsequent tasks (#1209).
     try {
-      const { pruneDeadProcesses } = await import("../bg-shell/process-manager.js");
+      const { pruneDeadProcesses, killSessionProcesses } = await import("../bg-shell/process-manager.js");
       pruneDeadProcesses();
+      killSessionProcesses();
     } catch {
       // Non-fatal
     }

